pydantic/fields.py

```diff
@@ -56,14 +56,14 @@ class _FromFieldInfoInputs(typing_extensions.TypedDict, total=False):
     description: str | None
     examples: list[Any] | None
     exclude: bool | None
-    gt: float | None
-    ge: float | None
-    lt: float | None
-    le: float | None
-    multiple_of: float | None
+    gt: annotated_types.SupportsGt | None
+    ge: annotated_types.SupportsGe | None
+    lt: annotated_types.SupportsLt | None
+    le: annotated_types.SupportsLe | None
+    multiple_of: annotated_types.MultipleOf | None
     strict: bool | None
-    min_length: int | None
-    max_length: int | None
+    min_length: annotated_types.MinLen | None
+    max_length: annotated_types.MaxLen | None
```

Comment:
For these, let's actually keep the original type (`float` and `int`) and add an additional union member as the `annotated_types` type. Ex:
`multiple_of: int | annotated_types.MultipleOf | None`

Reply:
@bjmc,

Thinking about this more, let's also add some tests accordingly, verifying that `MultipleOf` works in this case (it might not, due to expectations on the `pydantic-core` end of things.